### PR TITLE
fix: Improve shutdown procedure of ssh, portforward, wgtunnel cmds

### PR DESCRIPTION
**Pull request by mafredri**
We could turn it into a practice to wrap `cmd.Context()` so that we have
more fine-grained control of cancellation. Sometimes in tests we may be
running commands with a context that is never canceled.

Related to #3221

### CODE REVIEW

File: cli/ssh.go [hunk 1]
```diff
@@ -224,13 +236,17 @@ func ssh() *cobra.Command {
 
 			sshSession.Stdin = cmd.InOrStdin()
 			sshSession.Stdout = cmd.OutOrStdout()
-			sshSession.Stderr = cmd.OutOrStdout()
+			sshSession.Stderr = cmd.ErrOrStderr()
```
Comment (mafredri) [inline cli/ssh.go:239]: This is a drive-by change I did. It seemed wrong but perhaps I didn't understand the purpose?

**Review by kylecarbs — APPROVED** [review]
I'm in-between on whether this is desired or not, but I suppose it is. It seems weird to not use the command context directly, but then the caller (like our tests) is responsible for canceling it, which is even weirder I suppose.

File: cli/ssh.go (hunk 1 above)
Comment (kylecarbs) [reply cli/ssh.go:239]: Seems reasonable to me!

**Comment by mafredri** [comment]
> I'm in-between on whether this is desired or not, but I suppose it is. It seems weird to not use the command context directly, but then the caller (like our tests) is responsible for canceling it, which is even weirder I suppose.

We're still inheriting it, so in that sense we're still using it. I think this change can make more sense if we consider other scenarios, like adding tracing to the commands. We'd want to ensure that the context we're passing along contains the tracing information which means we shouldn't be accessing the command context directly.
